### PR TITLE
Issue#1035 - Assert Canonical Tag

### PR DESCRIPTION
**Pull request by juanmrad**
Add the same function to 2.0 flexible mink.

Add test to assert canonical tag exist and contains the proper link.
add appropriate needed function to assert any node contain attribute values.

### CODE REVIEW

File: src/Medology/Behat/Mink/FlexibleContext.php [hunk 1]
```diff
@@ -1272,6 +1272,37 @@ public function pressButton($locator, TraversableElement $context = null)
...
+     * @throws ExpectationException             If the nodes attributes do not match
+     * @throws UnsupportedDriverActionException When operation not supported by the driver
+     */
+    public function assertNodesHaveAttributeValues($locator, array $attributes, $selector = 'named', $occurrences = null)
+    {
+        /** @var NodeElement[] $links */
+        $nodes = $this->getSession()->getPage()->findAll($selector, $locator);
+        if (!count($nodes)) {
+            throw new ExpectationException("No node elements were found on the page using '$locator'", $this->getSession());
+        } elseif ($occurrences && count($nodes) != $occurrences) {
```
Comment (BKorenek) [inline src/Medology/Behat/Mink/FlexibleContext.php]: without an else this should probably just be two ifs.

Comment (BKorenek) [reply src/Medology/Behat/Mink/FlexibleContext.php]: Also ```!==``` instead of !=

File: web/assert-link-location.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+<html>
+
+<head>
+    <title>Text in Element for Flexible Mink</title>
+    <link rel="canonical" href="http://localhost.local/testing" />
+</head>
+
+<body>
+<div id="emptyDiv"></div>
+<div id="divWithText">This is a div</div>
```
Comment (BKorenek) [inline web/assert-link-location.html]: tab before each div.

File: src/Medology/Behat/Mink/LinkContext.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Medology\Behat\Mink;
```
Comment (BKorenek) [inline src/Medology/Behat/Mink/LinkContext.php]: this should be up top right?

File: web/assert-link-location.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <title>Text in Element for Flexible Mink</title>
+    <link rel="canonical" href="http://localhost.local/testing" />
+</head>
+
+<body>
+<div id="emptyDiv"></div>
```
Comment (BKorenek) [inline web/assert-link-location.html]: standard id convention is with dashes

Comment (juanmrad) [reply web/assert-link-location.html]: Using an already existing class :trollface:

**Pull request merged**